apps/go-mod-validator/src/github.ts

```diff
@@ -23,35 +23,62 @@ async function isCommitInDefaultBranch(
     head: commitSha,
   });
 
-  return status === "identical" || status === "behind";
+  const isInDefault = status === "identical" || status === "behind";
+  return {
+    isInDefault,
+    commitSha,
+  };
 }
 
 async function isTagInDefaultBranch(
   gh: Octokit,
   defaultBranch: string,
   mod: GoModuleWithTag,
-): Promise<boolean> {
-  const { data: repoTags } = await gh.rest.repos.listTags({
-    owner: mod.owner,
-    repo: mod.repo,
-  });
+): Promise<{
+  isInDefault: boolean;
+  commitSha: string;
+}> {
+  let commitSha = "";
+  try {
+    const tag = await gh.rest.git.getRef({
+      owner: mod.owner,
+      repo: mod.repo,
+      ref: `tags/${mod.tag}`,
+    });
 
-  for (const repoTag of repoTags) {
-    if (repoTag.name != mod.tag) {
-      continue;
+    // lightweight tags show up as "commit" type, annotated tags show up as "tag" type
+    const isAnnotatedTag = tag.data.object.type === "tag";
+    if (isAnnotatedTag) {
+      const getTagResp = await gh.rest.git.getTag({
+        owner: mod.owner,
+        repo: mod.repo,
+        tag_sha: tag.data.object.sha,
+      });
+      commitSha = getTagResp.data.object.sha;
```

Comment:
So it didn't support annotated tags before?

Reply:
It did, `git.getTag` only supports annotated tags but `repo.listTags` supports all tag types

Reply:
Then why is this logic required for annotated tags?

Reply:
we're not using `repo.listTags` anymore because its an inefficient API call. The only way to retrieve a single tag is to use the git database API instead which requires differentiation between the two tag types